src/baseline_automatic_pipeline_system/parse_okr_info.py

```diff
@@ -58,14 +61,104 @@ def parse_single_sentences(raw_sentences):
     # parse each tweet and store entity and proposition extractions
     parsed_sentences = {}
     for sent_id, sent in normed_sentences.iteritems():
-        try:
-            props_wrapper.parse(sent)
-            parsed_sentences[sent_id] = props_wrapper.get_okr()
-        except Exception as e:
-            logging.error("failed to parse sentence: " + sent +" | Error: " + str(e))
+        # try:
+        #     parsed_sentences[sent_id] = parse_tweet(sent)
+        # except Exception as e:
+        #     logging.error("failed to parse sentence: " + sent +" | Error: " + str(e))
+        parsed_sentences[sent_id] = parse_tweet(sent)
 
     return parsed_sentences
 
+def parse_tweet(tweet_text):
+    """ 
+    parse tweet - return a dict of tweet's propositions and entities.
+     We use SpaCy sentence segmentation to split the original tweet into grammatical sentences.
+     on each such sentence, we run the props_wrapper. Then, we accumulate the outputs to one output.
+    """
+    def props_parse(sent):
+        props_wrapper.parse(sent)
+        return props_wrapper.get_okr()
+
+    #return props_parse(tweet_text)
+
+    # use Spacy to segment tweet to sentences
+    doc = spacy_pipe(unicode(tweet_text))
+    # parse each sentence separately using props
+    props_parses = [props_parse(s.text) for s in doc.sents]
+    # return a merge of the parses
+    return combine_parses(doc, props_parses)
+
+def combine_parses(spacy_doc, props_parses):
```

Comment:
If there is a single sentence, maybe the sentence merging process can be skipped?
This seems like a time-costly procedure that will not be necessary in many tweets.